### PR TITLE
zpool: deduplicate root mount option

### PR DESCRIPTION
**Pull request by SuperSandro2000**

### CODE REVIEW

File: types/zpool.nix [hunk 1]
```diff
@@ -37,17 +37,6 @@
       default = [ "defaults" ];
       description = "Options to pass to mount";
     };
-    mountRoot = lib.mkOption {
```
Comment (Mic92) [inline types/zpool.nix:40]: @Lassulus where we would add mkRemovedOptionModule if we wanted to add them for submodules?

**Comment by Lassulus** [comment]
Do we want zfspools always to be at the root? What if people want to have tmpfs on /?

**Comment by SuperSandro2000** [comment]
This should just be for mounting the zpool in a live iso where it must be mounted at /mnt otherwise it kills the running OS.

**Comment by phaer** [comment]
> This should just be for relative mounting the zpool in a live iso where it must be mounted at /mnt otherwise it kills the running OS. It doesn't influence where the zpool is mounted in the resulting system.

But I couldn't build a system where I e.g. use `tmpfs` for `/mnt` while having a `zpool` at `/mnt/persist` with that?

**Review by Lassulus — APPROVED** [review]

**Comment by Lassulus** [comment]
applied in https://github.com/nix-community/disko/pull/294, will be merged tomorrow or so. thanks!

**Pull request closed without merging**